### PR TITLE
Read RJPEG Flir FFF Segment

### PR DESCRIPTION
**Pull request by rugheid**

### CODE REVIEW

File: src/buff-util.mjs [hunk 1]
```diff
@@ -41,6 +41,12 @@ export function getInt32(buffer, offset, littleEndian = false) {
 	else					return buffer.readInt32BE(offset)
 }
 
+export function getFloat32(buffer, offset, littleEndian = false) {
```
Comment (rugheid) [inline src/buff-util.mjs:44]: Parsing 32-bit floats wasn't implemented yet.

File: src/buff-util.mjs [hunk 2]
```diff
@@ -57,17 +63,22 @@ export function slice(buffer, start, end) {
...
 		} else {
 			var decoder = new TextDecoder('utf-8')
-			return decoder.decode(slice(buffer, start, end))
+			result = decoder.decode(slice(buffer, start, end))
 		}
 	} else {
-		return buffer.toString('ascii', start, end)
+		result = buffer.toString('ascii', start, end)
+	}
+	if (clean) {
```
Comment (rugheid) [inline src/buff-util.mjs:80]: The changes in this function allow cleaning the string by removing everything after the first null character.

File: src/parser.mjs [hunk 3]
```diff
@@ -81,6 +82,28 @@ function getExifSize(buffer, offset) {
...
+function isFlirFFFSegment(buffer, offset) {
+	return getUint32(buffer, offset + 4) === 0x464C4952 // 'FLIR'
+	// After FLIR, there always seems to be 00 01 00 0A, but I'm not sure so I'm not checking that yet
+	// 	&& getUint32(buffer, osset + 8) === 0x0001000A
+	// If this is followed by 'FFF\0' (46 46 46 00 in hex), this is an FFF segment
+		&& getUint32(buffer, offset + 12) === 0x46464600 // 'FFF\0'
+}
+
+function getFlirFFFSize(buffer, offset) {
+    // TODO: Implement size!
```
Comment (rugheid) [inline src/parser.mjs:98]: I'm not using size, but I'll still fix this TODO.

File: src/parser.mjs [hunk 4]
```diff
@@ -197,6 +220,7 @@ export class ExifParser extends Reader {
 		if (this.options.xmp)  this.parseXmpSegment()  // Additional XML data (in XML)
 		if (this.options.icc)  this.parseIccSegment()  // Image profile
 		if (this.options.iptc) this.parseIptcSegment() // Captions and copyrights
+		if (this.options.flir) this.parseFlirFFFSegment() // Additional data included by FLIR cameras
```
Comment (rugheid) [inline src/parser.mjs:223]: I currently require the options to pass `{ flir: true }`. Is this ok, or do you guys prefer to just automatically enable this in our fork?

File: src/parser.mjs [hunk 5]
```diff
@@ -444,6 +469,131 @@ export class ExifParser extends Reader {
...
+	}
+
+	parseFlirFFFRecordTags(flir, recordType, recordContentOffset, le) {
+	    if (!(recordType in tags.flir)) {
+			return
+		}
+
+		const recordTagsInfo = tags.flir[recordType];
+
+	    // TODO: What the fuck?!!
```
Comment (rugheid) [inline src/parser.mjs:554]: No idea where this comes from, but it works consistently, so I'm going to leave it for now. Once we have more photos I could investigate in more detail maybe.

File: src/parser.mjs (hunk 4 above)
Comment (BossensM) [reply src/parser.mjs:223]: Ok like this

File: src/buff-util.mjs (hunk 2 above)
Comment (BossensM) [reply src/buff-util.mjs:80]: ok

**Review by OlivierKamers — APPROVED** [review]

File: test/test.js [hunk 6]
```diff
@@ -357,6 +359,111 @@ describe('parser (exif data)', () => {
...
+				assert.equal(exif.flir.PlanckR2, 0.0441913828253746)
+				assert.equal(exif.flir.RawValueMedian, 3408)
+				assert.equal(exif.flir.RawValueRange, 3291)
+				assert.equal(exif.flir.FocusStepCount, 0)
+				assert.equal(exif.flir.FocusDistance, 0)
+			})
+
+			it(`003b.jpg`, async () => {
+				var exif = await parse(buffers['003b.jpg'], { flir: true, mergeOutput: false })
+                console.log(exif.flir)
```
Comment (BossensM) [inline test/test.js:415]: remove console?

File: test/test.js [hunk 7]
```diff
@@ -357,6 +359,111 @@ describe('parser (exif data)', () => {
...
+				assert.equal(exif.flir.PlanckO, -70)
+				assert.equal(exif.flir.PlanckR2, 0.0441913828253746)
+				assert.equal(exif.flir.RawValueMedian, 3408)
+				assert.equal(exif.flir.RawValueRange, 3291)
+				assert.equal(exif.flir.FocusStepCount, 0)
+				assert.equal(exif.flir.FocusDistance, 0)
+			})
+
+			it(`003b.jpg`, async () => {
+				var exif = await parse(buffers['003b.jpg'], { flir: true, mergeOutput: false })
```
Comment (BossensM) [inline test/test.js:414]:
```suggestion
				var exif = await parse(buffers['003b.jpg'], { flir: true })
```

Remove mergeOutput

**Review by BossensM — APPROVED** [review]

**Pull request merged**